supervisor/supervisor.go

```diff
@@ -275,8 +277,12 @@ func (s *Supervisor) Start() error {
 		"cpus":        s.machine.Cpus,
 	}).Debug("containerd: supervisor running")
 	go func() {
-		for i := range s.tasks {
-			s.handleTask(i)
+		for {
+			select {
+			case i := <-s.tasks:
```

Comment:
@mlaventure actually, I think the whole goroutine stuff for tasks is pretty useless as case b) in the docker issue isn't actually fixable as you described. What if we just add something like https://github.com/containerd/containerd/pull/852#discussion_r116355740 to tackle case c)? as you pointed out, channels aren't context aware and it's a client responsability to send a timeout. So, not fully sure what the parallelism here buys. Let me know.